R/adjust-probability-threshold.R

```diff
@@ -0,0 +1,106 @@
+#' Change the event threshold
+#'
+#' @param x A [container()].
+#' @param threshold A numeric value (between zero and one) or [hardhat::tune()].
+#' @examples
+#' library(dplyr)
+#' library(modeldata)
+#'
+#' post_obj <-
+#'   container(mode = "classification") %>%
+#'   adjust_probability_threshold(threshold = .1)
+#'
+#' two_class_example %>% count(predicted)
+#'
+#' post_res <- fit(
+#'   post_obj,
+#'   two_class_example,
+#'   outcome = c(truth),
+#'   estimate = c(predicted),
+#'   probabilities = c(Class1, Class2)
+#' )
+#'
+#' predict(post_res, two_class_example) %>% count(predicted)
+#' @export
+adjust_probability_threshold <- function(x, threshold = 0.5) {
+  check_container(x)
+  if (!is_tune(threshold)) {
+    check_number_decimal(threshold, min = 10^-10, max = 1 - 10^-10)
+  }
+
+  op <-
+    new_operation(
+      "probability_threshold",
+      inputs = "probability",
+      outputs = "class",
+      arguments = list(threshold = threshold),
+      results = list(trained = FALSE)
+    )
+
+  new_container(
+    mode = x$mode,
+    type = x$type,
+    operations = c(x$operations, list(op)),
+    columns = x$dat,
+    ptype = x$ptype,
+    call = current_env()
+  )
+}
+
+#' @export
+print.probability_threshold <- function(x, ...) {
+  # check for tune() first
+
+  if (is_tune(x$arguments$threshold)) {
+    cli_inform("Adjust probability threshold to optimized value.")
+  } else {
+    trn <- ifelse(x$results$trained, " [trained]", "")
+    cli_inform(c("Adjust probability threshold to  \\
+                    {signif(x$arguments$threshold, digits = 3)}{trn}"))
```

Comment:
I wouldn't clock what these error messages mean unless I saw the technical conditions that led up to them. Could we rewrite?